File: 1147-flip-columns-for-maximum-number-of-equal-rows/flip-columns-for-maximum-number-of-equal-rows.py
class Solution:
    def maxEqualRowsAfterFlips(self, matrix: List[List[int]]) -> int:
        lookup = {}
        rows = len(matrix)
        columns = len(matrix)
        count = 0

        for row in matrix:
            row_key = tuple(row)
            if (not row[0]):
                row_key = tuple([0 if char == 1 else 1 for char in row])
            
            if (row_key not in lookup):
                lookup[row_key] = 0
            lookup[row_key] += 1


        return max(lookup.values())

File: 1147-flip-columns-for-maximum-number-of-equal-rows/flip-columns-for-maximum-number-of-equal-rows.js
var maxEqualRowsAfterFlips = function(matrix) {
    let lookup = {}
    let result = 0

    for (let row of matrix) {
        let key = ""
        if (!row[0]) {
            for (let num of row) {
                key += num ? 0 : 1
            }
        }
        else {
            for (let num of row) key += num
        }
        if (!(key in lookup)) lookup[key] = 0
        lookup[key] += 1
        result = Math.max(result, lookup[key])

    }    
    return result
};